### PR TITLE
Add dockerfiles for pbench, collectd

### PR DESCRIPTION
**Pull request by chaitanyaenr**
This commit adds the dockerfiles, scripts needed to be build the
pbench-agent, pbench-controller, collectd images.

### CODE REVIEW

File: dockerfiles/pbench/agent/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    mv openshift*/oc /usr/local/bin && \
+    rm -rf openshift-origin-client-tools-*
+
+# Install ansible, other dependencies
+RUN mkdir -p /root/.ssh 
+RUN rpm -iUvh https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm && yum install -y ansible python-pip which
+RUN pip install requests
+
+# Setup sshd, initscripts
+RUN yum install -y openssh-server openssh-clients initscripts && yum clean all
```
Comment (sjug) [inline dockerfiles/pbench/agent/Dockerfile]: As above

File: dockerfiles/pbench/agent/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+RUN sed -i "s/#Port 22/Port 2022/" /etc/ssh/sshd_config
+RUN systemctl enable sshd
+EXPOSE 2022
+
+# Mount /proc
+COPY mount.sh /root/mount.sh
+RUN chmod +x /root/mount.sh
```
Comment (sjug) [inline dockerfiles/pbench/agent/Dockerfile]: As above

File: dockerfiles/pbench/controller/run.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+. /etc/profile.d/pbench-agent.sh
+mkdir -p /var/lib/pbench-agent/tools-default
+cp /root/.ssh/id_rsa /opt/pbench-agent/id_rsa
+sed -i "/^pbench_results_redirector/c pbench_results_redirector = ${pbench_server}" /opt/pbench-agent/config/pbench-agent.cfg
+sed -i "/^pbench_web_server/c pbench_web_server = ${pbench_server}"  /opt/pbench-agent/config/pbench-agent.cfg
+pbench-clear-tools
+ansible-playbook -i /root/inventory /root/pbench/contrib/ansible/openshift/pbench_register.yml
+if [[ "$?" == 0 ]]; then
```
Comment (sjug) [inline dockerfiles/pbench/controller/run.sh]: Could you add some new lines in between a bit, it's a bit easier to read vs one chunk.

**Comment by mffiedler** [comment]
In general, each RUN (each separate docker build directive for the most part) will result in a separate layer for the image.   We should combine RUNs as much as possible as @sjug suggests.

**Comment by chaitanyaenr** [comment]
@sjug @mffiedler minimizing the layers in order to reduce the size of the image makes perfect sense.  I had separate RUN commands to make the dockerfile more readable. I have made the changes requested PTAL.

File: dockerfiles/collectd/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,11 @@
+FROM fedora:latest
+MAINTAINER Naga Ravi Chaitanya Elluri <nelluri@redhat.com>
+
+# Install collectd and dependencies
+RUN dnf install -y git collectd collectd-disk procps && \
+    mkdir -p /root/collectd
```
Comment (sjug) [inline dockerfiles/collectd/Dockerfile]: Don't forget the `dnf clean all`

File: dockerfiles/pbench/controller/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    curl -L https://github.com/openshift/origin/releases/download/v1.2.1/openshift-origin-client-tools-v1.2.1-5e723f6-linux-64bit.tar.gz | tar -zx && \
+    mv openshift*/oc /usr/local/bin && \
+    rm -rf openshift-origin-client-tools-* && \
+    mv /opt/pbench-agent/config/pbench-agent.cfg.example /opt/pbench-agent/config/pbench-agent.cfg && \
+    sed -i "s/#Port 22/Port 2022/" /etc/ssh/sshd_config && \
+    systemctl enable sshd
+
+# setup pbench-ansible, svt    
+WORKDIR /root
+RUN git clone https://github.com/distributed-system-analysis/pbench.git && git clone https://github.com/openshift/svt.git
```
Comment (sjug) [inline dockerfiles/pbench/controller/Dockerfile]: Maybe line wrap this command as well, it's not the longest line (as there are above) but just easier to read at a glance.

**Comment by chaitanyaenr** [comment]
@sjug made the changes requested.

File: dockerfiles/pbench/controller/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,36 @@
+#Dockerfile for pbench-controller
+FROM centos/tools
+MAINTAINER Naga Ravi Chaitanya Elluri <nelluri@redhat.com>
+
+# Setup pbench, sshd, pbench-ansible, svt and install dependencies
+RUN rpm -iUvh https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm && \
```
Comment (sjug) [inline dockerfiles/pbench/controller/Dockerfile]: Missed this one `-i`/`-U`

Comment (chaitanyaenr) [reply dockerfiles/pbench/controller/Dockerfile]: Fixed.

**Comment by chaitanyaenr** [comment]
@sjug made the changes requested, PTAL.

**Comment by chaitanyaenr** [comment]
@sjug, @mffiedler thanks for the review. @mffiedler, @vikaslaad ready for merge?

**Comment by chaitanyaenr** [comment]
@mffiedler PTAL.

**Pull request merged**